lib/fizzy/parser.hpp

```diff
@@ -26,10 +26,12 @@ inline const uint8_t* skip(size_t num_bytes, const uint8_t* input, const uint8_t
 /// Parse `expr`, i.e. a function's instructions residing in the code section.
 /// https://webassembly.github.io/spec/core/binary/instructions.html#binary-expr
 ///
-/// @param input        The beginning of the expr binary input.
-/// @param end          The end of the binary input.
-/// @param have_memory  If the module (the context) contains imported or defined memory.
-parser_result<Code> parse_expr(const uint8_t* input, const uint8_t* end, bool have_memory);
+/// @param input          The beginning of the expr binary input.
+/// @param end            The end of the binary input.
+/// @param have_memory    If the module (the context) contains imported or defined memory.
+/// @param function_types Ordered types of functions in the module, including imported ones.
+parser_result<Code> parse_expr(const uint8_t* input, const uint8_t* end, bool have_memory,
+    const std::vector<FuncType>& function_types);
```

Comment:
I plan to try refactoring to bring the loop of splitting imports section by import kind from instantiation to parsing/validation, then here probably the entire `Module` will be passed instead of `have_memory` and `function_types`

Reply:
https://github.com/wasmx/fizzy/pull/268